wporg-gp-translation-events.php

```diff
@@ -366,24 +366,35 @@ function gp_event_nav_menu_items( array $items, string $location ): array {
 add_filter( 'gp_nav_menu_items', 'Wporg\TranslationEvents\gp_event_nav_menu_items', 10, 2 );
 
 /**
- * Generate a slug for the event post type when we save a draft event.
+ * Generate a slug for the event post type when we save a draft event or when we publish an event.
  *
- * Generate a slug based on the event title if it's not provided.
+ * Generate a slug based on the event title if:
+ * - The event is a draft.
+ * - The event is published and it was a draft just before.
  *
- * @param array $data An array of slashed post data.
+ * @param array $data    An array of slashed post data.
+ * @param array $postarr An array of sanitized, but otherwise unmodified post data.
  * @return array The modified post data.
  */
-function generate_event_slug( array $data ): array {
-	if ( 'event' === $data['post_type'] && 'draft' === $data['post_status'] ) {
-		if ( empty( $data['post_name'] ) ) {
+function generate_event_slug( array $data, array $postarr ): array {
+	if ( 'event' === $data['post_type'] ) {
+		if ( 'draft' === $data['post_status'] ) {
 			$data['post_name'] = sanitize_title( $data['post_title'] );
 		}
+		if ( 'publish' === $data['post_status'] ) {
+			if ( is_numeric( $postarr['ID'] ) && 0 !== $postarr['ID'] ) {
```

Comment:
I added the `0 !== $postarr['ID']` condition to make tests pass.

But this got me thinking: in the case where the event is not first saved as draft (i.e. user publishes directly), the `$data['post_name'] = sanitize_title( $data['post_title'] );` will never be called, right? I guess this is not what we want, but I'm not sure.

Reply:
> I added the `0 !== $postarr['ID']` condition to make tests pass.
> 
> But this got me thinking: in the case where the event is not first saved as draft (i.e. user publishes directly), the `$data['post_name'] = sanitize_title( $data['post_title'] );` will never be called, right? I guess this is not what we want, but I'm not sure.

If the event is not saved previously as a draft, the slug problems will be addressed by the [wp_insert_post](https://developer.wordpress.org/reference/functions/wp_insert_post/) function, so I think everything will be ok.

Reply:
Ah, that makes sense 👍